src/hyp3_sdk/hyp3.py

```diff
@@ -424,16 +424,21 @@ def prepare_insar_job(cls,
         return job_dict
 
     def submit_insar_isce_burst_job(self,
-                                    granule1: str,
-                                    granule2: str,
+                                    *args,
```

Comment:
I think you can be a bit simpler here; if users are using positional arguments, they can just use `reference` and `secondary` since it's type hinted as `Union[str, Iterable[str]]` and should be functionally equivalent. You really only care about maintaining `granule1, `granule2` for users using keyword arguments, so I would do this:

```python
    def submit_insar_isce_burst_job(self,
                                    reference: Union[str, Iterable[str]] = None,
                                    secondary: Union[str, Iterable[str]] = None,
                                    name: Optional[str] = None,
                                    apply_water_mask: bool = False,
                                    looks: Literal['20x4', '10x2', '5x1'] = '20x4',
                                    *,
                                    granule1: Optional[str] = None,
                                    granule2: Optional[str] = None,
                                    ) -> Batch:
```

Then after the function definition, you would check granules and reference/secondary; something like:

```python
if `reference` is None:
    if `granule1` is None:
        raise ValueError('Either reference and secondary must be provided, or granule1 and granule2')
    reference = granule1

if `secondary` is None:
    if `granule2` is None:
        raise ValueError('Either reference and secondary must be provided, or granule1 and granule2')
    secondary = granule2
```

and you can drop all of the `arguments = locals().copy()` stuff (there's no harm in a `granule1` and `granule2` variable existing).

---

Maybe not this PR, but in the future, it might be worth moving the `*` before `name`, forcing the options to come in as keyword arguments.